applications/exporter/application_round_result_exporter.py

```diff
@@ -0,0 +1,194 @@
+import csv
+from collections.abc import Iterable
+from dataclasses import asdict, dataclass
+from io import StringIO
+
+from django.db.models import Prefetch
+from django.http import FileResponse
+from lookup_property import L
+
+from applications.choices import ApplicationStatusChoice
+from applications.exporter.base_exporter import BaseCSVExporter, BaseExportRow
+from applications.models import AllocatedTimeSlot, ApplicationSection, ReservationUnitOption
+from applications.querysets.application_section import ApplicationSectionQuerySet
+from common.date_utils import local_date, local_date_string, local_time_string, local_timedelta_string
+
+
+@dataclass
+class ApplicationSectionExportRow(BaseExportRow):
+    # Application
+    application_id: str = ""
+    application_status: str = ""
+    applicant: str = ""
+
+    # Application Section
+    section_id: str = ""
+    section_status: str = ""
+    section_name: str = ""
+    reservations_begin_date: str = ""
+    reservations_end_date: str = ""
+    applied_reservations_per_week: str = ""
+    reservation_min_duration: str = ""
+    reservation_max_duration: str = ""
+
+    # Reservation Unit
+    reservation_unit_name: str = ""
+    unit_name: str = ""
+    day_of_the_week: str = ""
+    begin_time: str = ""
+    end_time: str = ""
+
+    # Price
+    price: str = ""  # TODO
+
+
+class ApplicationRoundResultCSVExporter(BaseCSVExporter):
+    """
+    Usage:
+    >>> exporter = ApplicationRoundResultCSVExporter(application_round_id=1)
+    >>> csv_export = exporter.export()
+    >>> print(csv_export.getvalue())
+    """
+
+    application_round_id: int
+
+    def __init__(self, application_round_id: int):
+        self.application_round_id = application_round_id
+        self.max_options = 0
+        self.options_part_template = []
+
+    def export(self) -> StringIO | None:
+        sections: ApplicationSectionQuerySet = self._get_queryset()
+        if not sections:
+            return None
+
+        csv_file = StringIO()
+        csv_writer = csv.writer(csv_file)
+
+        # Write header rows
+        csv_writer.writerow(self._get_header_rows())
+
+        # Write data rows
+        for section in sections:
+            section_data = self._get_single_row_data(section)
+            for row in section_data:
+                csv_writer.writerow(row)
+
+        return csv_file
+
+    def export_as_file_response(self) -> FileResponse | None:
+        csv_file = self.export()
+        if csv_file is None:
+            return None
+
+        today = local_date()
+        file_name = f"application_round_results_{self.application_round_id}_{today.isoformat()}.csv"
+        response = FileResponse(csv_file.getvalue(), content_type="text/csv")
+        response["Content-Disposition"] = f"attachment;filename={file_name}"
+        return response
```

Comment:
Same as this: https://github.com/City-of-Helsinki/tilavarauspalvelu-core/pull/1157/files#r1570004182